BUILD

```diff
@@ -1,6 +1,35 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_binary")
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+load("@bazel_gazelle//:def.bzl", "gazelle")
 
+# Run `bazel run //:gazelle` each time you add a new directory.
+# As a result, `BUILD` files will be generated for you within that directory.
+# gazelle:prefix github.com/slsa-framework/example-package
+gazelle(name = "gazelle")
+
+# Run `bazel run //:gazelle-update-repos` each time `go.mod` changes (e.g., you
+# add a new dependency). This would result in Bazel workspace dependencies also
+# being brought up to date.
+gazelle(
+    name = "gazelle-update-repos",
+    args = [
+        "-from_file=go.mod",
+        "-to_macro=deps.bzl%go_dependencies",
+        "-prune",
+    ],
+    command = "update-repos",
+)
+
+# Existing rules (you can write your own or use the ones generated by `gazelle`).
 go_binary(
     name = "hello",
+    embed = [":example-package_lib"],
+    visibility = ["//visibility:public"],
+)
+
+go_library(
+    name = "example-package_lib",
     srcs = ["main.go"],
+    importpath = "github.com/slsa-framework/example-package",
+    visibility = ["//visibility:private"],
+    deps = ["@com_github_pborman_uuid//:uuid"],
 )
```

Comment:
Is this how gazelle generates the rules? I think you can combine these into one `go_binary` rule, but I guess it leaves open the ability to write tests over the library.

Reply:
Sadly, yes, that's how the rules are generated. We could remove the setup and revert this to the original format (if we don't remove gazelle then next time it runs it will create the same diff). I couldn't find a way to tell it to not touch this file anymore.

Reply:
no worries. I think this is ok as it is. I was just curious if you did it this way for a specific reason.